src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
-
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue'
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (c >= a + b || b >= a + c || a >= b + c) {
+      throw new Error(`The longest side of a triangle 
+        shouldn't be less than a sum of two others`);
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The figure`s values should be more than 0');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```

Comment:
```suggestion
    const p = (this.a + this.b + this.c) / 2;
```

Reply:
it's p in Math

Reply:
it is S in math

Reply:
when you declare P as a perimeter => much logically to name half of perimeter like 'p' 
as for radiuses when we have circumcircle radius it's named R, incenter radius => r
But for this case it doesn't matter. I agree, half of people using `s`, so no worries about it:)